utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -154,12 +154,15 @@ def uses_host_tests(self):
         """
         return True
 
-    def swift_flags(self, args):
+    def swift_flags(self, args, resource_path=None):
```

Comment:
This method is currently used to generate Swift cross-compilation flags for `build-script-impl` products alone, but I was able to reuse it for `Testing` by passing in a separate `resource_path`. That `resource_path` calculation can only be done externally, where the particular build product is known.